### PR TITLE
curriculum planning - stage 1

### PR DESCRIPTION
**Pull request by mrhm-dev**
I have completed the initial course curriculum planning. This curriculum needs more polish but I will move forward after you approve the initial planning. After finishing the course curriculum completely I will move forward with course content planning, resource collection, code examples, and assignments.

### CODE REVIEW

File: Outline.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Outline - Build Modern APIs in NodeJS
+
+Welcome to the course, in this course we are going to learn how to build modern APIs using NodeJS. Mainly we will focus on REST API as it will be found all over the world. But we will also try to understand other popular competitors of REST APIs like GraphQL, JSON API, gRPC like techniques. End of this course you will learn - 
+
+-   What is API (Application Programming Interface)
+-   What is REST API (Representation State Transfer)
+-   Why do We Need API
+-   Best Alternatives of REST API and Their Use Cases
+-   How to Implement REST API using Node and Express
+-   How to Implement GraphQL API using GraphQL Yoga or Express GraphQL
```
Comment (kizggerg) [inline Outline.md:10]: I've also seen Apollo GraphQL as a popular framework in this space. 
We should base our GraphQL framework choice based on which one is most searched as that is likely to draw in the most customers.

File: Outline.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+
+## Lessons
+
+I will update describe every chapter later after your confirmation. 
+
+
+
+1.  Introduction
+    1.  Course Introduction (Slide)
```
Comment (kizggerg) [inline Outline.md:39]: We should be sure to add a "what this course is NOT" slide / video to indicate up front that this is not going to be a full stack / database course.

File: Outline.md (hunk 1 above)
Comment (mrhm-dev) [reply Outline.md:10]: You may please check these links - 
https://dev.to/vinodchauhan7/which-is-best-apollo-server-vs-express-graphql-vs-graphql-yoga-vs-prisma-vs-apollo-server-express-5g42
https://dev.to/yeeiodev/differences-between-express-graphql-apollo-and-graphql-yoga-3m82

File: Outline.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    2.  Understand PassportJS (Screen)
+    3.  Implement PassportJS Local Strategy (Screen)
+10.  REST API Best Practices
+     1.  Open API Specifications (Screen)
+     2.  API Best Practices (Slide)
+11.  API Versioning and Documentation
+     1.  How to Manage Version of API (Screen)
+     2.  Introduction to Swagger (Screen)
+     3.  Implement Swagger in Our Application (Screen)
+12.  API Testing using JEST & Supertest
```
Comment (kizggerg) [inline Outline.md:99]: We should include some content here on the test pyramid: https://martinfowler.com/articles/practical-test-pyramid.html#:~:text=The%20%22Test%20Pyramid%22%20is%20a,put%20it%20into%20practice%20properly.

Particularly relevant here is:
1. API System Test - Call the actual API in its actual environment, no mocking whatsoever. In my experience this is typically done with Jest + Node Fetch calls.
2. API Service Test - Call the API, but mocking all external dependencies (e.g. other APIs being called). In my experience this is done with Jest + Supertest (as you outline here)
3. Unit Tests - Calls functions and methods directly, testing internal functionality (provides safety when refactoring). In my experience this is typically done in Jest alone.

File: Outline.md (hunk 1 above)
Comment (mrhm-dev) [reply Outline.md:10]: You may please check these links -
https://dev.to/vinodchauhan7/which-is-best-apollo-server-vs-express-graphql-vs-graphql-yoga-vs-prisma-vs-apollo-server-express-5g42
https://dev.to/yeeiodev/differences-between-express-graphql-apollo-and-graphql-yoga-3m82

File: Outline.md (hunk 2 above)
Comment (mrhm-dev) [reply Outline.md:39]: Sure, I will add that

File: Outline.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+15.  Create Resolvers for Review
+     1.  Create Schema for Review (Screen)
+     2.  Create Query Resolvers for Review (Screen)
+     3.  Create Mutation Resolvers for Review (Screen)
+16.  Add Authentication to GraphQL
+     1.  Authentication Technique for GraphQL (Screen)
+     2.  Implement PassportJS Local Authentication Strategy (Screen)
+17.  GraphQL Best Practices
+     1.  GraphQL Best Practices (Slide)
+18.  What Next
```
Comment (kizggerg) [inline Outline.md:124]: Is there any other API technologies you think we should focus on here? I've seen gRPC becoming more and more popular for those who do everything on Google Cloud. 

Perhaps we can have a section on "up and coming technologies" that does a _very simple_ walkthrough through other API technologies - what do you think?

**Review by kizggerg — APPROVED** [review]
Everything is looking good from here.
Feel free to merge when you feel you have enough details from this course outline to begin the next stage of preparing an "end-to-end" lesson 😃

File: Outline.md (hunk 4 above)
Comment (mrhm-dev) [reply Outline.md:124]: sounds good. we can add that too.

File: Outline.md (hunk 3 above)
Comment (mrhm-dev) [reply Outline.md:99]: Ok, I will keep that in mind

**Pull request merged**